### PR TITLE
gogs: 0.11.91 -> 0.12.3

### PR DESCRIPTION
**Pull request by braunse**
###### Motivation for this change

Upstream release: https://github.com/gogs/gogs/releases/tag/v0.12.3
FIxes:
 * https://nvd.nist.gov/vuln/detail/CVE-2020-15867
 * https://nvd.nist.gov/vuln/detail/CVE-2020-14958
 * https://nvd.nist.gov/vuln/detail/CVE-2020-9329
 * https://nvd.nist.gov/vuln/detail/CVE-2018-15178

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by braunse** [comment]
Result of `nixpkgs-review pr 102061` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gogs</li>
  </ul>
</details>

**Comment by braunse** [comment]
Closure size comparison:

    /nix/store/lffwq0nv4djkbwmxsv0a67z84q2ql8bf-gogs-0.11.91          448391248  # before
    /nix/store/4isb92hw8ivgw1s00wj6qsn94wfw1vxb-gogs-0.12.3           444369544 # after

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102061` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gogs</li>
  </ul>
</details>

File: pkgs/applications/version-management/gogs/default.nix [hunk 1]
```diff
@@ -34,18 +35,17 @@ buildGoPackage rec {
     (  optional sqliteSupport "sqlite"
     ++ optional pamSupport "pam");
 
-  outputs = [ "out" "data" ];
+  # outputs = [ "out" "data" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/gogs/default.nix:38]:
```suggestion
```

File: pkgs/applications/version-management/gogs/default.nix [hunk 2]
```diff
@@ -34,18 +35,17 @@ buildGoPackage rec {
...
+  outputs = [ "out" ];
 
   postInstall = ''
-    mkdir $data
-    cp -R $src/{public,templates} $data
+    # mkdir $data
+    # cp -R $src/{public,templates} $data
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/gogs/default.nix:43]:
```suggestion
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102061` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gogs</li>
  </ul>
</details>

File: pkgs/applications/version-management/gogs/default.nix [hunk 3]
```diff
@@ -34,18 +35,14 @@ buildGoPackage rec {
     (  optional sqliteSupport "sqlite"
     ++ optional pamSupport "pam");
 
-  outputs = [ "out" "data" ];
+  outputs = [ "out" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/gogs/default.nix:38]: IIRC this is the default and does not need to be defined but please test.

**Comment by SuperSandro2000** [comment]
Please squash the commits together.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102061` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gogs</li>
  </ul>
</details>

**Review by stigtsp — APPROVED** [review]
LGTM

Result of `nixpkgs-review pr 102061` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gogs</li>
  </ul>
</details>

Tested OK on `linux x64`:
- Enabled via `service.gogs`
- Setup using `service.gogs.useWizard` with sqlite3 backend
- Created repo, pushed commit via http

**Comment by dotlambda** [comment]
Can we backport the update without making a breaking change in the NixOS module?

**Pull request merged**